README.rst

```diff
@@ -1,334 +1,503 @@
+   default value.
 
-To take advantage of the *datafiles* fixture in a test function, add
-*datafiles* as one of the test function parameters (per usual with `pytest`_
-fixtures) and decorate the test function with *@pytest.mark.datafiles(file1,
-file2, dir1, dir2, ...)*. See the examples below.
+*  **on_duplicate:** Specify the action to take when duplicate
+   files/directories are found. Possible values are: *exception*,
+   *ignore* and *replace*. The default value is *exception*.
 
-The *datafiles* variable in your test function is a py.path object
-(`tmpdir`_) where the copied files are located. Under Linux systems this
-will most likely be some subdirectory of */tmp/*.
+   *  *exception:* An exception is raised instead of copying the
+      duplicate file/directory.
 
+   *  *ignore:* The second (or subsequent) files/directories with the
+      same name as the first one are simply ignored (i.e., the first
+      file/directory with the duplicate name is kept).
 
-Options
--------
-
-The following options can be specified as keyword arguments (kwargs) to the
-*@pytest.mark.datafiles* decorator function:
-
-- **keep_top_dir:** For all parameters that represent directories, keep that
-  directory instead of only (recursively) copying its content. Possible values
-  are *True* or *False*. *False* is the default value.
-- **on_duplicate:** Specify the action to take when duplicate files/directories
-  are found. Possible values are: *exception*, *ignore* and *replace*. The
-  default value is *exception*.
-
-  - *exception:* An exception is raised instead of copying the duplicate
-    file/directory.
-  - *ignore:* The second (or subsequent) files/directories with the same name
-    as the first one are simply ignored (i.e., the first file/directory with the
-    duplicate name is kept).
-  - *replace:* The second (or subsequent) files/directories with the same name
-    replace the previous ones (i.e., the last file/directory with the duplicate
-    name is kept).
+   *  *replace:* The second (or subsequent) files/directories with the
+      same name replace the previous ones (i.e., the last
+      file/directory with the duplicate name is kept).
 
 See below for some *examples*.
 
 
 Installation
-------------
+============
+
+.. code:: bash
+
+   pip install pytest-datafiles
+
+
+Upgrade to 3.0
+==============
+
+Version 3 now uses `tmp_path
+<https://docs.pytest.org/en/latest/tmp_path.html>`_, resulting in
+*pathlib.Path* objects instead of *py.path*.
+
+Your tests may need to be adjusted.
+
+.. code:: python
+
+   """Upgrade to v3."""
+   import os
+
+   import pytest
+   import py
+
 
-.. code-block:: bash
+   @pytest.mark.datafiles('/tmp/big_files/film1.mp4')
+   def test_convert_to_pypath(datafiles):
+       """Convert `datafiles` to `py.path` in place.
 
-    pip install pytest-datafiles
+       Changing the type back to py.path - should be avoided!
+
+       Better to update the tests to make use of `pathlib`.
```

Comment:
If I understand you correctly you are saying the following and I suggest rephrasing it like this:

```python
"""
If your tests are taking advantage of py.path the simplest change you can make
is at the top of each test convert the pathlib.Path object back to a py.path .

Note that it would be better to migrate to pathlib.Path, as shown in the next
example, because py.path is deprecated.
"""
```

Reply:
done